doc/source/cluster/kubernetes/examples/distributed-checkpointing-with-gcsfuse.md

```diff
@@ -0,0 +1,296 @@
+
+## Why Distributed Checkpointing with GCSFuse?
+
+In large-scale, high-performance machine learning, distributed checkpointing is crucial for fault tolerance, ensuring that if a node fails during training, the process can be resumed from the latest saved checkpoint instead of starting from scratch.
+While it's possible to directly reference remote storage paths (e.g., `gs://my-checkpoint-bucket`), using Google Cloud Storage FUSE (GCSFuse) offers distinct advantages for distributed applications. GCSFuse allows you to mount Cloud Storage buckets like local file systems, making checkpoint management more intuitive for distributed applications that rely on these semantics. Furthermore, GCSFuse is designed for high-performance workloads, delivering the performance and scalability needed for distributed checkpointing of large models.
+
+[Distributed checkpointing](https://docs.ray.io/en/latest/train/user-guides/checkpoints.html), in combination with [GCSFuse](https://cloud.google.com/storage/docs/gcs-fuse), allows for larger-scale model training with increased availability and efficiency.
+
+## Create a Kubernetes cluster on GKE
+
+Create a GKE cluster with the [GCSFuse CSI driver](https://cloud.google.com/kubernetes-engine/docs/how-to/persistent-volumes/cloud-storage-fuse-csi-driver) and [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) enabled, as well as a GPU node pool with 4 L4 GPUs:
+```
+export PROJECT_ID=<your project id>
+gcloud container clusters create kuberay-with-gcsfuse \
+    --addons GcsFuseCsiDriver \
+    --cluster-version=1.29.4 \
+    --location=us-east4-c \
+    --machine-type=g2-standard-8 \
+    --release-channel=rapid \
+    --num-nodes=4 \
+    --accelerator type=nvidia-l4,count=1,gpu-driver-version=latest \
+    --workload-pool=${PROJECT_ID}.svc.id.goog
+```
+
+Verify the successful creation of your cluster with 4 GPUs:
+```
+$ kubectl get nodes "-o=custom-columns=NAME:.metadata.name,GPU:.status.allocatable.nvidia\.com/gpu"
+NAME                                                  GPU
+gke-kuberay-with-gcsfuse-default-pool-xxxx-0000       1
+gke-kuberay-with-gcsfuse-default-pool-xxxx-1111       1
+gke-kuberay-with-gcsfuse-default-pool-xxxx-2222       1
+gke-kuberay-with-gcsfuse-default-pool-xxxx-3333       1
+```
+
+## Install the KubeRay operator
+
+Follow [Deploy a KubeRay operator](kuberay-operator-deploy) to install the latest stable KubeRay operator from the Helm repository.
+The KubeRay operator Pod must be on the CPU node if you set up the taint for the GPU node pool correctly.
+
+## Configuring the GCS Bucket
+
+Create a GCS bucket that will be used as the remote filesystem.
+```
+BUCKET=<your GCS bucket>
+gcloud storage buckets create gs://$BUCKET --uniform-bucket-level-access
+```
+
+Create a Kubernetes ServiceAccount that will be used to grant the RayCluster access to mount the GCS bucket:
```

Comment:
```suggestion
Create a Kubernetes ServiceAccount that KubeRay uses to grant the RayCluster access to mount the GCS bucket:
```

Reply:
We follow the Google Developer Documentation Style Guide (https://developers.google.com/style), which discourages the use of future tense.